foundry.md

```diff
@@ -561,7 +641,62 @@ Utils
          </account>
 ```
 
- - selectors for cheat code functions.
+- `#setPrank NEWCALLER` will set the `<prank/>` subconfiguration for the given account.
+
+```k
+    syntax KItem ::= "#setPrank" Int [klabel(foundry_setPrank)]
+ // -----------------------------------------------------------
+    rule <k> #setPrank NEWCALLER => . ... </k>
+         <callDepth> CD </callDepth>
+         <caller> CL </caller>
+         <origin> OG </origin>
+         <prank>
+           <prevCaller> .Account => CL </prevCaller>
+           <prevOrigin> .Account => OG </prevOrigin>
+           <newCaller> _ => NEWCALLER </newCaller>
+           <active> false => true </active>
+           <depth> _ => CD </depth>
+           <singleCall> _ => false </singleCall>
+           ...
+         </prank>
+```
+
+- `#setPrankWithOrigin NEWCALLER NEWORIGIN` will set the `<prank/>` subconfiguration for the given account.
+
+```k
+    syntax KItem ::= "#setPrankWithOrigin" Int Int [klabel(foundry_setPrankWithOrigin)]
+ // -----------------------------------------------------------------------------------
+    rule <k> #setPrankWithOrigin NEWCALLER NEWORIGIN => . ... </k>
+         <callDepth> CD </callDepth>
+         <caller> CL </caller>
+         <origin> OG </origin>
+         <prank>
+           <prevCaller> .Account => CL </prevCaller>
+           <prevOrigin> .Account => OG </prevOrigin>
+           <newCaller> _ => NEWCALLER </newCaller>
+           <newOrigin> _ => NEWORIGIN </newOrigin>
+           <active> false => true </active>
+           <depth> _ => CD </depth>
+           <singleCall> _ => false </singleCall>
+         </prank>
+```
+
+- `#endPrank` will end the prank, restoring the previous caller and origin to the `<caller>` and `<origin>` cells in the configuration.
+
+```k
+    syntax KItem ::= "#endPrank" [klabel(foundry_endPrank)]
+ // -------------------------------------------------------
```

Comment:
Similarly, what happens if you do `#endPrank` when `<active> false </active>`? Maybe we should have a rule which goes to `#foundry_error "No prank started"` instead.

Also, what about case where you do a `#startPrank ACCT`, then an `#endPrank`? It seems to me that you'll end up overriding the `<origin>` cell with `.Account`, because this rule doesn't check if `<prevOrigin>` is `.Account`. Maybe we need a rule to handle `<proveOrigin> .Account </prevOrigin>` case?

Reply:
Ah wait! I misread the semantics! It appears that you already are handling this smoothly by having separate `<newOrigin>` and `<prevOrigin>` cell. Nice.